basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/principal/decoder/ElytronPrincipalDecoderTestCase.java

```diff
@@ -0,0 +1,686 @@
+                        .addAll(constantPrincipalDecoder3name, constantPrincipalDecoder1name)
+                        .build());
+        } finally {
+            ops.removeIfExists(aggregatePrincipalDecoderAddress);
+            ops.removeIfExists(constantPrincipalDecoder1address);
+            ops.removeIfExists(constantPrincipalDecoder2address);
+            ops.removeIfExists(constantPrincipalDecoder3address);
+            adminOps.reloadIfRequired();
+        }
+    }
+
+    /**
+     * @tpTestDetails Create Elytron Aggregate Principal Decoder instance in model
+     * and try to remove it in Web Console's Elytron subsystem configuration.
+     * Validate the resource is not any more visible in Aggregate Principal Decoder table.
+     * Validate created resource is not any more present in the model.
+     */
+    @Test
+    public void removeAggregatePrincipalDecoderTest() throws Exception {
+        String
+            aggregatePrincipalDecoderName = randomAlphanumeric(5),
+            constantPrincipalDecoder1name = randomAlphanumeric(5),
+            constantPrincipalDecoder2name = randomAlphanumeric(5),
+            constant1value = randomAlphanumeric(5),
+            constant2value = randomAlphanumeric(5);
+        Address
+            aggregatePrincipalDecoderAddress =
+                elyOps.getElytronAddress(AGGREGATE_PRINCIPAL_DECODER, aggregatePrincipalDecoderName),
+            constantPrincipalDecoder1address =
+                elyOps.getElytronAddress(CONSTANT_PRINCIPAL_DECODER, constantPrincipalDecoder1name),
+            constantPrincipalDecoder2address =
+                elyOps.getElytronAddress(CONSTANT_PRINCIPAL_DECODER, constantPrincipalDecoder2name);
+        ResourceVerifier aggregatePrincipalDecoderVerifier = new ResourceVerifier(aggregatePrincipalDecoderAddress, client);
+
+        try {
+            ops.add(constantPrincipalDecoder1address, Values.of(CONSTANT, constant1value)).assertSuccess();
+            ops.add(constantPrincipalDecoder2address, Values.of(CONSTANT, constant2value)).assertSuccess();
+            ops.add(aggregatePrincipalDecoderAddress, Values.of(PRINCIPAL_DECODERS, new ModelNodeListBuilder()
+                    .addAll(constantPrincipalDecoder1name, constantPrincipalDecoder2name)
+                    .build())).assertSuccess();
+            aggregatePrincipalDecoderVerifier.verifyExists();
+
+            page.navigateToDecoder().selectResource(AGGREGATE_PRINCIPAL_DECODER_LABEL).getResourceManager()
+                    .removeResource(aggregatePrincipalDecoderName).confirmAndDismissReloadRequiredMessage().assertClosed();
+            assertFalse("Removed resource should not be present in the table any more!",
+                    page.resourceIsPresentInMainTable(aggregatePrincipalDecoderName));
+            aggregatePrincipalDecoderVerifier.verifyDoesNotExist();
+        } finally {
+            ops.removeIfExists(aggregatePrincipalDecoderAddress);
```

Comment:
Shouldn't `constantPrincipalDecoder1address` and `constantPrincipalDecoder2address` be removed here to?